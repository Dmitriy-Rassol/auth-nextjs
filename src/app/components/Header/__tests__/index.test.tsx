import mockRouter from 'next-router-mock';
import { render, screen } from '@testing-library/react';
import { Header } from '../index';

jest.mock('next/navigation', () => require('next-router-mock'));

// jest.mock('next/navigation', () => {
//     return {
//         useRouter: () => {}
//     }
// });

describe('Header', () => {
  it('renders the component', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});