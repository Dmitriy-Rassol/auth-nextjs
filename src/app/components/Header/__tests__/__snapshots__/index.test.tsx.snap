// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the component 1`] = `
<div>
  <header>
    <nav
      class="bg-white px-2 py-2.5 dark:border-gray-700 dark:bg-gray-800 sm:px-4"
    >
      <div
        class="mx-auto flex flex-wrap items-center justify-between container"
      >
        <a
          class="flex items-center"
          href="https://flowbite.com/"
        >
          <img
            alt="Flowbite Logo"
            class="mr-3 h-6 sm:h-9"
            src="https://flowbite.com/docs/images/logo.svg"
          />
          <span
            class="self-center whitespace-nowrap text-xl font-semibold dark:text-white"
          >
            Flowbite
          </span>
        </a>
        <button
          class="inline-flex items-center rounded-lg p-2 text-sm text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600 md:hidden"
          data-testid="flowbite-navbar-toggle"
        >
          <span
            class="sr-only"
          >
            Open main menu
          </span>
          <svg
            aria-hidden="true"
            class="h-6 w-6 shrink-0"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </button>
        <div
          class="w-full md:block md:w-auto hidden"
          data-testid="flowbite-navbar-collapse"
        >
          <ul
            class="mt-4 flex flex-col md:mt-0 md:flex-row md:space-x-8 md:text-sm md:font-medium"
          >
            <a
              aria-current="page"
              class="hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-white md:dark:hover:text-blue-500 dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
              href="/"
            >
              Home
            </a>
            <a
              class="hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-white md:dark:hover:text-blue-500 dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
              href="/login"
            >
              Login
            </a>
            <a
              class="hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-white md:dark:hover:text-blue-500 dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
              href="/register"
            >
              Register
            </a>
          </ul>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
